pipeline_utils/lib/yaml_parser.py

```diff
@@ -0,0 +1,414 @@
+#!/usr/bin/env python
+
+import yaml
+
+
+###############################################################
+#   Functions
+###############################################################
+def load_yaml(file):
+    """
+        return a generator to loaded yaml documents in file
+    """
+    with open(file) as stream:
+        try:
+            for d in yaml.safe_load_all(stream):
+                yield d
+        except yaml.YAMLError as exc:
+            sys.exit(exc)
+
+
+###############################################################
+#   YamlWfl, Yaml Workflow
+###############################################################
+class YamlWfl(object):
+
+    def __init__(self, yaml_d):
+        """
+        """
+        for key, val in yaml_d.items():
+            setattr(self, key, val)
+        self._validate()
```

Comment:
See my comment on handling YAML/JSON in `to_json` (you should just store the loaded dictionary and convert directly).

Reply:
I am actually not doing YAML/JSON 1:1. I am trying to simplify the YAML as much as possible and better organize some tag names in a way that makes more sense to a potential users. Our JSONs currently are using tags that are non-intuitive and are not harmonized between objects, also how to structure them is not as clear if you are not us. I am trying to make it more compact and simple and user friendly. The parser is basically reading the YAML (that should be more user streamlined now) and picking the information it needs to then create our non intuitive JSON objects.